### PR TITLE
 python3Packages.ducc0: init at 0.23.0

### PR DESCRIPTION
**Pull request by phiadaarr**
###### Description of changes

Add `ducc0` as a new package to nix.

Homepage: https://gitlab.mpcdf.mpg.de/mtr/ducc
Documentation: https://mtr.pages.mpcdf.de/ducc

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by phiadaarr** [comment]
@FRidh @jonringer I think I need some help. I do not know why "Label PR / labels (pull_request_target)" fails. Could you give me a hint, what might be going wrong?

**Comment by fabaff** [comment]
> @FRidh @jonringer I think I need some help. I do not know why "Label PR / labels (pull_request_target)" fails. Could you give me a hint, what might be going wrong?

There was a temporary issue.

File: pkgs/development/python-modules/ducc0/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{ lib, buildPythonPackage, fetchgit, pytest, pybind11, numpy }:
+
+buildPythonPackage rec {
+  pname = "ducc0";
+  version = "0.23.0";
+  myversion = "0_23_0";
+
```
Comment (fabaff) [inline pkgs/development/python-modules/ducc0/default.nix:7]: Please add `  disabled = pythonOlder "3.7";`

See https://gitlab.mpcdf.mpg.de/mtr/ducc/-/blob/ducc0/setup.py#L138

File: pkgs/development/python-modules/ducc0/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  src = fetchgit {
```
Comment (fabaff) [inline pkgs/development/python-modules/ducc0/default.nix:8]: There is [`fetchFromGitLab`](https://nixos.org/manual/nixpkgs/stable/#fetchfromgitlab) could be used here to get the source.

File: pkgs/development/python-modules/ducc0/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    url = "https://gitlab.mpcdf.mpg.de/mtr/ducc";
+    rev = "ducc0_" + myversion;
+    sha256 = "dOc3TihtoRuLfniC9zQ1MEbnvGJHzFZfZ8+J8Dnw6Lk=";
+  };
+
+  buildInputs = [ pybind11 ];
+  propagatedBuildInputs = [ numpy ];
+
+  checkInputs = [ pytest ];
```
Comment (fabaff) [inline pkgs/development/python-modules/ducc0/default.nix:17]: Please use `pytestCheckHook` if possible.

File: pkgs/development/python-modules/ducc0/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  pythonImportsCheck = [ "ducc0" ];
+
+  meta = with lib; {
+    homepage = "https://gitlab.mpcdf.mpg.de/mtr/ducc";
+    description = "Efficient algorithms for Fast Fourier transforms and more";
+    license = licenses.gpl2Plus;
+    maintainers = with maintainers; [ parras ];
+  };
+
+}
```
Comment (fabaff) [inline pkgs/development/python-modules/ducc0/default.nix:29]:
```suggestion
  };
}
```

File: pkgs/development/python-modules/ducc0/default.nix (hunk 1 above)
Comment (phiadaarr) [reply pkgs/development/python-modules/ducc0/default.nix:6]: done.

File: pkgs/development/python-modules/ducc0/default.nix (hunk 2 above)
Comment (phiadaarr) [reply pkgs/development/python-modules/ducc0/default.nix]: done.

File: pkgs/development/python-modules/ducc0/default.nix (hunk 3 above)
Comment (phiadaarr) [reply pkgs/development/python-modules/ducc0/default.nix]: deon.

File: pkgs/development/python-modules/ducc0/default.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  pname = "ducc0";
+  version = "0.23.0";
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.mpcdf.mpg.de";
+    owner = "mtr";
+    repo = "ducc";
+    rev = "ducc0_" + builtins.replaceStrings ["."] ["_"] version;
```
Comment (phiadaarr) [inline pkgs/development/python-modules/ducc0/default.nix:13]: @fabaff thanks for your helpful comments. Apart from integrating these, I also updated this line here. I hope it is fine.

Comment (SuperSandro2000) [reply pkgs/development/python-modules/ducc0/default.nix:13]:
```suggestion
    rev = "ducc0_${lib.replaceStrings ["."] ["_"] version}";
```

**Pull request merged**